### PR TITLE
 Patching SystemJS.delete so that hot reloading can work

### PR DESCRIPTION
**Pull request by internettrans**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-      - "5.3.0"
+      - "node"
```
Comment (internettrans) [inline .travis.yml:3]: latest version of node (fixed the build for me)

File: src/system-script.js [hunk 2]
```diff
@@ -17,6 +17,23 @@ window.define = window.canopyDefine = function(name, deps, m) {
 
 window.define.amd = true;
 
+const ogSystemDelete = outerSystem.delete;
+
+outerSystem.delete = function(normalizedName) {
+  // This is asynchronous, but SystemJS.delete is synchronous. We just accept the race condition :'(
+  SystemJS.import('sofe').then(sofe => sofe.locate({address: normalizedName})).then(url => {
```
Comment (internettrans) [inline src/system-script.js]: This is pretty canopy-specific -- wasn't sure if that was a problem or not for this project.

File: src/system-script.js [hunk 3]
```diff
@@ -1,81 +1,101 @@
 let scriptNameMap = window.__systemAmdScript.scriptNameMap;
 let outerSystem = SystemJS;
 
+const ogSystemDelete = outerSystem.delete;
+
+outerSystem.delete = function(normalizedName) {
```
Comment (internettrans) [inline src/system-script.js:6]: This is new

File: test/system-script.test.js [hunk 4]
```diff
@@ -1,114 +1,133 @@
...
+            .catch(fail);
+          }, false);
+        script.addEventListener('error', fail, false);
+        script.src = '/base/test/fixtures/module-preloaded.js';
+        head.appendChild(script);
+      })
+      .catch(fail);
+  });
+
+  it("Should delete a module's script tag and remove from scriptNameMap when SystemJS.delete is called", function(done) {
```
Comment (internettrans) [inline test/system-script.test.js:115]: This is new

**Review by blittle — APPROVED** [review]
🐫

**Pull request merged**